From df8a44fea53819042700c028ae0259aa52600688 Mon Sep 17 00:00:00 2001
From: INAJIMA Daisuke <inajima@soum.co.jp>
Date: Tue, 2 Jul 2019 19:00:06 +0900
Subject: [PATCH] Fix size of OMX_U32 and OMX_S32 on aarch64

OMX_x32 is apparently expected as 32-bit long, but they are defined as
unsigned/sined long that are 64-bit length on aarch64.  H.264 decode by
OMX failed due to this mismatch because proprietary OMX library treats
it as 32-bit long, but Firefox treats it as 64-bit long.
---
 media/openmax_il/il112/OMX_Types.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/media/openmax_il/il112/OMX_Types.h b/media/openmax_il/il112/OMX_Types.h
index 873ba7a48676..c3957e6e09d2 100644
--- a/media/openmax_il/il112/OMX_Types.h
+++ b/media/openmax_il/il112/OMX_Types.h
@@ -142,10 +142,10 @@ typedef unsigned short OMX_U16;
 typedef signed short OMX_S16;
 
 /** OMX_U32 is a 32 bit unsigned quantity that is 32 bit word aligned */
-typedef unsigned long OMX_U32;
+typedef unsigned int OMX_U32;
 
 /** OMX_S32 is a 32 bit signed quantity that is 32 bit word aligned */
-typedef signed long OMX_S32;
+typedef signed int OMX_S32;
 
 
 /* Users with compilers that cannot accept the "long long" designation should
-- 
2.22.0

